### PR TITLE
The official docs are good, let's remove any specific debug-toolbar code / hacks we have.

### PR DESCRIPTION
**Pull request by EnTeQuAk**

### CODE REVIEW

**Comment by diox** [comment]
r+

**Comment by muffinresearch** [comment]
As mentioned, you need to turn off CSP to make the debug toolbar work which was covered by the special settings file removed here. If people are going to need to do that in local dev separately we'll need an "anti-footgun" test to make sure report only mode doesn't get committed.

**Comment by EnTeQuAk** [comment]
I rebased to master, updated docs and tests. r?

File: src/olympia/amo/tests/test_middleware.py [hunk 1]
```diff
@@ -83,15 +83,6 @@ def test_message(self):
...
-def test_hide_password_middleware():
-    request = RequestFactory().post('/', dict(x=1, password=2, password2=2))
-    request.POST._mutable = False
-    ScrubRequestOnException().process_exception(request, Exception())
-    assert request.POST['x'] == '1'
-    assert request.POST['password'] == '******'
-    assert request.POST['password2'] == '******'
-
-
 class TestNoDjangoDebugToolbar(TestCase):
```
Comment (diox) [inline src/olympia/amo/tests/test_middleware.py]: We can probably remove this test as well.

**Comment by diox** [comment]
r+wc

File: src/olympia/amo/tests/test_middleware.py (hunk 1 above)
Comment (EnTeQuAk) [inline src/olympia/amo/tests/test_middleware.py]: I actually re-added it to make sure we never do something stupid.

**Pull request merged**